node/pkg/fetcher/fetcher.go

```diff
@@ -221,47 +222,107 @@ func (f *App) insertRdb(ctx context.Context, name string, value float64) error {
 	return db.Set(ctx, key, string(data), time.Duration(5*time.Minute))
 }
 
-func (f *App) fetch(fetcher Fetcher) ([]float64, error) {
-	adapterFeeds := fetcher.Feeds
+func (a *App) fetch(fetcher Fetcher) ([]float64, error) {
+	feeds := fetcher.Feeds
 
 	data := []float64{}
+	dataChan := make(chan float64)
+	errChan := make(chan error)
+
+	var wg sync.WaitGroup
+	wg.Add(len(feeds))
+
+	for _, feed := range feeds {
+		go func(feed Feed) {
+			defer wg.Done()
+			definition := new(Definition)
+			err := json.Unmarshal(feed.Definition, &definition)
+			if err != nil {
+				errChan <- err
+				return
+			}
+			res, err := a.requestFeed(*definition)
+			if err != nil {
+				errChan <- err
+				return
+			}
 
-	for _, feed := range adapterFeeds {
-		definition := new(Definition)
-		err := json.Unmarshal(feed.Definition, &definition)
-		if err != nil {
-			log.Error().Err(err).Msg("failed to unmarshal feed definition")
-			continue
-		}
-		res, err := utils.GetRequest[interface{}](definition.Url, nil, definition.Headers)
-		if err != nil {
-			log.Error().Err(err).Msg("failed to get request")
-			continue
-		}
+			result, err := utils.Reduce(res, definition.Reducers)
+			if err != nil {
+				errChan <- err
+				return
+			}
 
-		result, err := utils.Reduce(res, definition.Reducers)
-		if err != nil {
-			log.Error().Err(err).Msg("failed to reduce")
-			continue
-		}
+			dataChan <- result
+		}(feed)
+	}
+
+	go func() {
```

Comment:
I am not sure it is necessary to make a go-routine for waiting until everybody is finished and closing channels. It could be more idiomatic to use defer for it 🤔 But maybe I missed something.

Reply:
Also, it could be better to define it closer to the line 233.

Reply:
yh I also think it's an overkill, i'll remove wait groups and close channel with defer block